### PR TITLE
custom render inputs and adjust the placement of the day picker based…

### PR DESCRIPTION
**Pull request by Zacthor**
… on item selected

### CODE REVIEW

File: src/components/DateRangePickerInput.jsx [hunk 1]
```diff
@@ -224,70 +289,24 @@ function DateRangePickerInput({
         isRTL && styles.DateRangePickerInput__rtl,
         !noBorder && styles.DateRangePickerInput__withBorder,
         block && styles.DateRangePickerInput__block,
-        showClearDates && styles.DateRangePickerInput__showClearDates,
-      )}
-    >
+        showClearDates && styles.DateRangePickerInput__showClearDates
```
Comment (jkslyby) [inline src/components/DateRangePickerInput.jsx:292]: Should this be present for custom rendering as well?

File: src/components/DateRangePickerInput.jsx [hunk 2]
```diff
@@ -297,11 +316,11 @@ function DateRangePickerInput({
             styles.DateRangePickerInput_clearDates,
             small && styles.DateRangePickerInput_clearDates__small,
             !customCloseIcon && styles.DateRangePickerInput_clearDates_default,
-            !(startDate || endDate) && styles.DateRangePickerInput_clearDates__hide,
+            !(startDate || endDate) &&
```
Comment (jkslyby) [inline src/components/DateRangePickerInput.jsx:319]: Same thing here - do we need to implement our own UI to call the clear function for custom rendering?

**Review by jkslyby — APPROVED** [review]

File: src/components/DateInput.jsx [hunk 3]
```diff
@@ -232,7 +234,10 @@ class DateInput extends React.PureComponent {
           value={value}
           onChange={this.onChange}
           onKeyDown={this.onKeyDown}
-          onFocus={onFocus}
+          onFocus={e => {
+            setInputHeight(e.target.getBoundingClientRect().y - FANG_HEIGHT_PX - 3);
+            onFocus(e);
+          }}
```
Comment (ellisio) [inline src/components/DateInput.jsx:240]: Looks like they have methods on the class. To be consistent, can we have a `this.onFocus`?

File: src/components/DateRangePicker.jsx [hunk 4]
```diff
@@ -291,6 +293,11 @@ class DateRangePicker extends React.PureComponent {
     this.container = ref;
   }
 
+  setInputHeight(h) {
+    console.log('set', h)
```
Comment (ellisio) [inline src/components/DateRangePicker.jsx:297]: Console statement

File: src/components/DateRangePicker.jsx [hunk 5]
```diff
@@ -458,7 +465,8 @@ class DateRangePicker extends React.PureComponent {
       theme: { reactDates },
     } = this.props;
 
-    const { dayPickerContainerStyles, isDayPickerFocused, showKeyboardShortcuts } = this.state;
+    const { dayPickerContainerStyles, isDayPickerFocused, showKeyboardShortcuts, inputHeight } = this.state;
+    console.log('state', inputHeight)
```
Comment (ellisio) [inline src/components/DateRangePicker.jsx:469]: console statement

File: src/components/DateRangePickerInput.jsx [hunk 6]
```diff
@@ -178,43 +178,108 @@ function DateRangePickerInput({
   small,
   regular,
   styles,
+  renderInputs,
+  setInputHeight
 }) {
+  console.log('render');
```
Comment (ellisio) [inline src/components/DateRangePickerInput.jsx:184]: console statement

**Review by ellisio — APPROVED** [review]

**Review by kmiller9393 — APPROVED** [review]

**Pull request closed without merging**